src/Symfony/Component/DependencyInjection/ContextualizedContainerBuilderInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+/**
+ * Implemented by a ContainerBuilder sharing a context during build time.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+interface ContextualizedContainerBuilderInterface
```

Comment:
I thought about this too. It's not so easy as we are trying to avoid considering the `ContainerBuilder` as a `Container`, having `ContainerBuilder` in the name of this interface may help in this goal, nothing except a `ContainerBuilder` should implement this. @ogizanagi your thoughts?